lib/new_relic/agent/instrumentation/thread.rb

```diff
@@ -0,0 +1,26 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require_relative 'thread/chain'
+require_relative 'thread/prepend'
+
+DependencyDetection.defer do
+  named :thread
+
+  depends_on do
+    defined?(Thread)
```

Comment:
I think I'd yank out `depends_on` entirely (or make it a noop like `depends_on; true; end` and also smoosh the two `executes` blocks together.

Reply:
Would it make sense to use `depends_on` to toggle the instrumentation for the Thread-related configuration, like I think it's done in `lib/new_relic/agent/instrumentation/active_record_notifications.rb`? I wonder if it might be more performant than using a conditional? (Though it may also require some refactoring...)

Reply:
@kaylareopelle the way the dependency detection works actually automatically checks the config for us (for anything using instrumentation.something), that way we don't need to add a depends on block to check it. I think that file you mentioned has the configs there because the configs are not the same format because it has been around longer than the `instrumentation.something` configs. That file also has 2 depends on blocks, the other one is checking to see if things are defined that need to be.
But also yeah, there is no reason Thread wouldn't be defined right? since it's part of ruby itself. We probably don't need the depends on block to check that then.